modules/light-clients/08-wasm/light_client_module_test.go

```diff
@@ -0,0 +1,285 @@
+package wasm_test
+
+import (
+	"encoding/json"
+	"time"
+
+	wasmvm "github.com/CosmWasm/wasmvm"
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+
+	wasmtesting "github.com/cosmos/ibc-go/modules/light-clients/08-wasm/testing"
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v8/modules/core/23-commitment/types"
+	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
+	ibcerrors "github.com/cosmos/ibc-go/v8/modules/core/errors"
+	"github.com/cosmos/ibc-go/v8/modules/core/exported"
+	ibctm "github.com/cosmos/ibc-go/v8/modules/light-clients/07-tendermint"
+	ibctesting "github.com/cosmos/ibc-go/v8/testing"
+)
+
+const (
+	tmClientID   = "07-tendermint-0"
+	wasmClientID = "08-wasm-100"
+)
+
+func (suite *WasmTestSuite) TestRecoverClient() {
+	var (
+		expectedClientStateBz               []byte
+		subjectClientID, substituteClientID string
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expErr   error
+	}{
+		// TODO(02-client routing): add successful test when light client module does not call into 08-wasm ClientState
```

Comment:
can be uncommented?

Reply:
needs issue: likely needs mock vm wired up (at a glance)